.github/workflows/diff-javadoc.yml

```diff
@@ -0,0 +1,58 @@
+name: Diff Javadoc
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Make Dir
+        run: mkdir ~/diff
+
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: temurin
+          cache: gradle
+
+      - name: Build
+        run: ./gradlew kotlindoc
```

Comment:
I think it may make sense to generate docs only for products that changed to avoid flakes and speed up this job. there is an example on how it's done in another workflow https://github.com/firebase/firebase-android-sdk/blob/00ea7ee68c9660e3d56c502b3f3df83026fda8df/.github/workflows/health-metrics.yml#L126-L129

And https://github.com/firebase/firebase-android-sdk/blob/00ea7ee68c9660e3d56c502b3f3df83026fda8df/.github/workflows/ci_tests.yml#L33-L34

wdyt?

Reply:
Health metrics does not use a matrix and instead uses a modules.json directly in the same action. So I think something along those lines can be done here as well, i.e.

1. checkout PR
2. run writeChangedProjects
3. parse the modules into something like `CHANGED=":firebase-common,firebase-components"` and then run `./gradlew {CHANGED}:kotlindoc` on both branches

Reply:
I've run into an issue where a lot of our SDK projects don't support the `kotlinDoc` task and are bricking the entire build when I try to run them, and I don't have a clean way to filter them out to only build the tasks that do, thinking it might make sense to revisit this optimization in the future as it's taking more time than I expected, thoughts?